src/resolution_functions/models/model_base.py

```diff
@@ -187,3 +161,58 @@ def citation(self) -> list[str]:
             The citation.
         """
         return self._citation
+
+    def _validate_settings(self,
+                           model_data: ModelData,
+                           settings: dict[str, int | float | None]
+                           ) -> dict[str, int | float]:
+        """
+        Validates the user-provided `settings` against the models restrictions and fills in defaults
+
+        Parameters
+        ----------
+        model_data
+            The data associated with the model for a given version of a given instrument.
+        settings
+            The user-provided :term:`settings<setting>`
+
+        Returns
+        -------
+        validated_settings
+            The user-provided settings that comply with the model's restrictions and with any `None`
+            values replaced with defaults.
+        """
+        out = {}
+        for name, setting in settings.items():
+            if setting is None:
+                try:
+                    out[name] = model_data.defaults[name]
+                except KeyError:
+                    raise InvalidInputError(f'Model "{type(self).__name__}" does not have a default'
+                                            f' value for the "{name}" setting, so one must be '
+                                            f'provided by the user.')
+                continue
+
+            try:
+                restriction = model_data.restrictions[name]
+            except KeyError:
+                out[name] = setting
+                continue
+
+            if isinstance(restriction, list):
+                if len(restriction) == 2:
+                    if not restriction[0] <= setting <= restriction[1]:
```

Comment:
How would I specify a restriction that is >0, but unbounded? Is there an easy way to specify `inf`?

Reply:
`val: !!float inf` works for infinity

Reply:
This seems a great example of "make the regular case easy and the exotic case possible".

If several models with Inf bounds come along, it could be worth special-casing. Until then, let's not.